### PR TITLE
[#IOCIT-9] Updated profile handlers to accept the new optional field from the profile model 

### PR DESCRIPTION
**Pull request by arcogabbo**
This PR depends on #252 

#### List of Changes

Updated `io-functions-commons` dependency to take the new model.
Updated `io-ts-commons` dependency to take `Semver` type.
Updated the profile handlers tests 
Updated profile handlers to correctly manipulate the new field

#### Motivation and Context

Correctly accept the optional field

#### How Has This Been Tested?

Build and tests passed

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: CreateProfile/__tests__/handler.test.ts
Comment (arcogabbo) [inline CreateProfile/__tests__/handler.test.ts:1]: I removed the linter ignoring comments because the tests file are ignored by eslint.

File: UpdateProfile/__tests__/handler.test.ts
Comment (BurnedMarshal) [inline UpdateProfile/__tests__/handler.test.ts:851]: This should never happen. The decode of the `findLastVersionByModelId` model method adds the default value.

File: UpdateProfile/__tests__/handler.test.ts
Comment (BurnedMarshal) [inline UpdateProfile/__tests__/handler.test.ts:897]: `lastAppVersion` override is not needed here, `aRetrievedProfile` already has it.
```suggestion
      findLastVersionByModelId: jest.fn(() => TE.of(some(aRetrievedProfile))),
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

**Review by BurnedMarshal — APPROVED** [review]
LGTM apart a minor change.
Wait for #252 to be merged before merging this pull request.

File: openapi/index.yaml
Comment (BurnedMarshal) [inline openapi/index.yaml:545]: To avoid nested type import we should import the `AppVersion` type in `index.yaml.template`

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

**Pull request merged**
